custom_components/tahoma/tahoma_device.py

```diff
@@ -109,14 +105,45 @@ def device_info(self) -> Dict[str, Any]:
         manufacturer = self.select_state(CORE_MANUFACTURER_NAME_STATE) or "Somfy"
         model = self.select_state(CORE_MODEL_STATE) or self.device.widget
 
-        return {
-            "identifiers": {(DOMAIN, self.unique_id)},
-            "manufacturer": manufacturer,
-            "name": self.name,
-            "model": model,
-            "sw_version": self.device.controllable_name,
+        name = self.name
+        device_url = self.device_url
+
+        # Some devices, such as the Smart Thermostat have several devices in one physical device,
+        # with same device url, terminated by '#' and a number.
+        # We use the base device url as the device identifier.
+        if "#" in self.device_url:
+            device_url, _ = self.device.deviceurl.split("#")
+
+            if not self.device_url.endswith("#1"):
+                entity_registry = self.hass.data["entity_registry"]
+                name = next(
+                    (
+                        entry.original_name
+                        for entity_id, entry in entity_registry.entities.items()
+                        if entry.unique_id == f"{device_url}#1"
+                    ),
+                    None,
+                )
+
+        device_info = {
+            "identifiers": {(DOMAIN, device_url)},
         }
 
+        # Only return device identifier for sub devices.
+        if "#" in self.device_url and not self.device_url.endswith("#1"):
+            return device_info
```

Comment:
Why not return the other information? It introduces weird behavior?

Reply:
HA does use the device info from the last entity of a device, while we prefer to show information of the first device..

Reply:
The following information are not the same across the devices?
```
{
                "name": name,
                "manufacturer": manufacturer,
                "model": model,
                "sw_version": self.device.controllable_name,
            }
```

Reply:
Not sure... Controllable name is different per Somfy entity for sure (io: xxx).